Documentation/cmdref/cilium-agent.md

```diff
@@ -24,14 +24,14 @@ cilium-agent [flags]
       --blacklist-conflicting-routes                  Don't blacklist IP allocations conflicting with local non-cilium routes (default true)
       --bpf-compile-debug                             Enable debugging of the BPF compilation process
       --bpf-ct-global-any-max int                     Maximum number of entries in non-TCP CT table (default 262144)
-      --bpf-ct-global-tcp-max int                     Maximum number of entries in TCP CT table (default 1000000)
+      --bpf-ct-global-tcp-max int                     Maximum number of entries in TCP CT table (default 524288)
       --bpf-ct-timeout-regular-any duration           Timeout for entries in non-TCP CT table (default 1m0s)
       --bpf-ct-timeout-regular-tcp duration           Timeout for established entries in TCP CT table (default 6h0m0s)
       --bpf-ct-timeout-regular-tcp-fin duration       Teardown timeout for entries in TCP CT table (default 10s)
       --bpf-ct-timeout-regular-tcp-syn duration       Establishment timeout for entries in TCP CT table (default 1m0s)
       --bpf-ct-timeout-service-any duration           Timeout for service entries in non-TCP CT table (default 1m0s)
       --bpf-ct-timeout-service-tcp duration           Timeout for established service entries in TCP CT table (default 6h0m0s)
-      --bpf-nat-global-max int                        Maximum number of entries for the global BPF NAT table (default 841429)
+      --bpf-nat-global-max int                        Maximum number of entries for the global BPF NAT table (default 524288)
```

Comment:
@borkmann @brb ^^^ do you know what the impact would be?

Given that the value is not configurable using Helm, maybe it's better to leave `bpf-nat-global-max` at the existing value for now and adjust it in a separate PR if needed?

Reply:
We've sized the SNAT tables by default a bit smaller than the CT tables and similarly it should be done here (cc @brb ). Concern is that given SNAT cleanup depends on CT GC right now, then once CT LRU evicts entries, they go out of reach for SNAT GC. Hence 1mio vs 841429.

Reply:
Thanks for the explanation. So I guess we should add a way to configure `bpf-nat-global-max` using Helm?

Reply:
Yes, I think it would be good.

Reply:
Will send a separate PR to allow configuring `bpf-nat-global-max` using Helm.

Reply:
Sent #10511 which will introduce the Helm option but leaving the value intact (i.e. at 841429) for now.

Since the value of `bpf-nat-global-max` is derived from `bpf-ct-global-tcp-max` and thus changed in this PR, I'll change the default value of `bpf-nat-global-max` for Helm in this PR as well.

Reply:
I didn't get a clear picture of what the datapath impact is here, but I'm presuming that we need similar upgrade docs handling for this option as for the CT TCP map option?

Reply:
Upgrade docs added in latest revision